src/api/event/controllers/event.js

```diff
@@ -1,9 +1,38 @@
-'use strict';
+const { createCoreController } = require('@strapi/strapi').factories;
 
-/**
- * event controller
- */
+module.exports = createCoreController('api::event.event', ({ strapi }) => ({
+  async find(ctx) {
+    // filter with the query
+    const { entityId, entityType } = ctx.query;
 
-const { createCoreController } = require('@strapi/strapi').factories;
+    // Fetch only the events that match the filters
+    const filters = {};
+    if (entityId) filters.entityId = entityId;
+    if (entityType) filters.entityType = entityType;
+
+    const events = await strapi.entityService.findMany('api::event.event', {
+      filters,  // Apply filters to fetch specific events
+      populate: ['eventUser'],  // Populate relations (e.g., eventUser)
+    });
+
+    // Retrieve latest entityName for each event Dynamically
+    const latestEvents = await Promise.all(events.map(async (event) => {
+      let entityName = '';
+
+      if (event.entityType === 'IdeaCard') {
+        const ideaCard = await strapi.entityService.findOne('api::idea-card.idea-card', event.entityId);
+        entityName = ideaCard ? ideaCard.ideaName : 'Unknown Idea';
+      } else if (event.entityType === 'Comment') {
```

Comment:
Hey @oriyomibadmus just a minor comment here.

This `else` is necessary ? Can we remove that and get the same result ?

```
else if (event.entityType === 'Comment')
```

Now, in order to catch other scenarios, another question, `event.entityType` can be != 'IdeaCard' and != 'Comment' ?